src/layouts/BasicLayout/index.tsx

```diff
@@ -2,20 +2,27 @@ import { Layout } from "antd";
 
 import Header from "@/components/Header";
 import Footer from "@/components/Footer";
+import type { User } from "@/types/user";
 
 import styles from "./index.module.less";
 
 const { Content } = Layout;
 
 export type BasicLayoutProps = {
   children: React.ReactNode;
+  user?: User | null;
+  location?: string;
 };
 
-const BasicLayout: React.FC<BasicLayoutProps> = (props) => {
+const BasicLayout: React.FC<BasicLayoutProps> = ({
+  user = null,
+  location = "",
+  children,
+}) => {
   return (
     <Layout className={styles.root}>
-      <Header />
-      <Content>{props.children}</Content>
+      <Header location={location} user={user} />
```

Comment:
prop `withLogo` could be provided from `BasicLayout` props.

it could be `loggedIn` prop, for example.